samples/hello-oboe/src/main/cpp/HelloOboeEngine.cpp

```diff
@@ -34,25 +34,30 @@
  * - Calculating the audio latency of the stream
  *
  */
-HelloOboeEngine::HelloOboeEngine(): mLatencyCallback(std::make_unique<LatencyTuningCallback>(*this)) {
+HelloOboeEngine::HelloOboeEngine()
+        : mLatencyCallback(std::make_unique<LatencyTuningCallback>(*this)) {
     start();
-    updateLatencyDetection();
 }
 
 double HelloOboeEngine::getCurrentOutputLatencyMillis() {
-    if (!mIsLatencyDetectionSupported) return -1;
+    if (!mIsLatencyDetectionSupported) return -1.0;
+
+    // Save local copy of shared pointer while we measure latency.
+    std::shared_ptr<oboe::AudioStream> localStream = mStream;
```

Comment:
@mnaganov Would mind taking a look at this? We are struggling to understand whether the copying of the `shared_ptr` is necessary.

Reply:
I think maybe I should use lock(mLock) here and then I will not need to copy the shared_ptr.

Reply:
Been thinking some more about this. The issue here is the assumption that `mStream` will always point to a valid stream object, and in a state where `getTimestamp` and the other methods for latency detection make sense. That isn't necessarily true: 

- `createPlaybackStream` may not have been called yet
- The current playback device may have been unplugged (e.g. headphones) leaving the stream in an indeterminate state of closing/closed/deleted
- The user may have changed the playback device in the drop down so the stream is (by user request) being closed and another stream being restarted, during which time latency detection will be unavailable (although the method will still currently be being called since it's on a separate JVM thread using `TimerTask`). 

This is a common problem: accessing streams and stream properties from multiple threads so it'd be good to sit down and thrash out all the scenarios (especially the stream lifecycle) and how to handle them correctly so we can document/update code samples throughout.

Reply:
We do not assume the stream is open. See line 47.

Reply:
At the moment HelloOboe is broken. This CL fixes the main problems. Can we make more improvements in the next CL?

Reply:
I switched it to use a lock instead of localstream.